src/main/kotlin/com/terraformation/backend/funder/db/FundingEntityUserStore.kt

```diff
@@ -1,18 +1,35 @@
 package com.terraformation.backend.funder.db
 
+import com.terraformation.backend.customer.model.requirePermissions
 import com.terraformation.backend.db.default_schema.UserId
 import com.terraformation.backend.db.funder.FundingEntityId
+import com.terraformation.backend.db.funder.tables.references.FUNDING_ENTITIES
 import com.terraformation.backend.db.funder.tables.references.FUNDING_ENTITY_USERS
+import com.terraformation.backend.funder.model.FundingEntityModel
 import jakarta.inject.Named
 import org.jooq.DSLContext
 
 @Named
 class FundingEntityUserStore(private val dslContext: DSLContext) {
   fun getFundingEntityId(userId: UserId): FundingEntityId? {
+    requirePermissions { readUser(userId) }
+
     return dslContext
         .select(FUNDING_ENTITY_USERS.FUNDING_ENTITY_ID)
         .from(FUNDING_ENTITY_USERS)
         .where(FUNDING_ENTITY_USERS.USER_ID.eq(userId))
         .fetchOne(FUNDING_ENTITY_USERS.FUNDING_ENTITY_ID)
   }
+
+  fun fetchEntityByUserId(userId: UserId): FundingEntityModel? {
+    requirePermissions { readUser(userId) }
+
+    return dslContext
+        .select(FUNDING_ENTITIES.asterisk())
+        .from(FUNDING_ENTITIES)
+        .join(FUNDING_ENTITY_USERS)
+        .on(FUNDING_ENTITY_USERS.FUNDING_ENTITY_ID.eq(FUNDING_ENTITIES.ID))
+        .where(FUNDING_ENTITY_USERS.USER_ID.eq(userId))
+        .fetchOne { FundingEntityModel.of(it) }
+  }
```

Comment:
There may be a way that I'm less familiar with for combining these 2 methods to a common private method. I wasn't sure with the join. Let me know if these should be combined